### ISSUE TITLE
libMonoPosixHelper.dll is missing in Windows installer

### ISSUE THREAD

**Issue by localuser2** [mono/mono#7439]
## Steps to Reproduce

1. Download Mono 5.10.0.14 Windows installer
2. Install it under Windows

## Current Behavior

INSTALLDIR\bin\libMonoPosixHelper.dll is missing. This results in problems with applications which are utilizing the Mono.Posix namespace

## Expected Behavior

INSTALLDIR\bin\libMonoPosixHelper.dll is available after installation

### On which platforms did you notice this

[ ] macOS
[ ] Linux
[ x] Windows

**Version Used**:

```
Mono JIT compiler version 5.10.0 (Visual Studio built mono)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           normal
        SIGSEGV:       normal
        Notification:  Thread + polling
        Architecture:  x86
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        GC:            sgen (concurrent by default)
```

**Comment by akoeplinger** [comment]
I do see the file in `C:\Program Files (x86)\Mono\bin\MonoPosixHelper.dll` with Mono 5.10.0.140 from http://www.mono-project.com/download/stable/#download-win

Could you please check which .msi you used?

**Comment by localuser2** [comment]
Yea. Of course I mean "libMonoPosixHelper.dll". Sorry for the irritation.

**Comment by localuser2** [comment]
It is the current stable Installer for Mono 5.10.0.140

**Comment by akoeplinger** [comment]
That's expected, the library doesn't have the `lib` prefix on Windows.

**Comment by localuser2** [comment]
But it was present in recent versions, Mono 5.4.x for instance (I did not try Mono 5.8). If I run a Gtk# application with internationalization in Mono 5.10, the message strings in the GUI are inconsistent (mix of English and the current system language). If the dll (`libMonoPosixHelper.dll` from Mono 5.4) is copied into the bin-directory, it is working as expected.

In earlier Mono versions both libraries (`MonoPosixHelper.dll` and `libMonoPosixHelper.dll`) have been present. If it is necessary, I can provide a small code sample to clarify the problem.

**Comment by akoeplinger** [comment]
That'd be helpful, afaik `libMonoPosixHelper.dll` was included by accident in earlier releases and nothing should use it.

**Comment by luhenry** [comment]
On top of that `libMonoPosixHelper.dll` is to access posix APIs from managed, and this sounds quite useless on Windows.

**Comment by localuser2** [comment]
I can deliver an code example tomorrow. In the meantime, maybe this helps:

Some magic seems to happen in this method: [Mono.Unix.Native.Stdlib.VersionCheck()](https://github.com/mono/mono/blob/mono-5.10.0.140/mcs/class/Mono.Posix/Mono.Unix.Native/Stdlib.cs#L403). If the property `versionCheckPerformed` is set to be always true, the problem disappears. However I do not understand the idea of the code.

https://github.com/mono/mono/blob/mono-5.10.0.140/mcs/class/Mono.Posix/Mono.Unix.Native/Stdlib.cs#L403

**Comment by akoeplinger** [comment]
@luhenry MonoPosixHelper isn't useless on Windows it has a few methods that work fine there.

@localuser2 the version check is to ensure that we're using a compatible version of the library.

**Comment by localuser2** [comment]
Okay. I will provide an example tomorrow. In any case: Thanks for the quick help. I am impressed! :+1:

**Comment by jonpryor** [comment]
Additionally, `Stdlib.VersionCheck()` calls `Stdlib.VersionStringPtr()`, which P/Invokes into `Stdlib.MPH`, which is `MonoPosixHelper`. Thus, it *should* be hitting `MonoPosixHelper.dll` on Windows, *not* `libMonoPosixHelper.dll`.

**Comment by localuser2** [comment]
It seems to be not that simple. The problem does not to occur if I use Mono 5.10 with the included Gtk#. We use a modified Gtk# which  (according to "Process Monitor") does load `libMonoPosixHelper.dll`.

The confusing fact is that the symptoms are the same if I
- Start the application with our modified Gtk# and without `libMonoPosixHelper.dll`
- Start the application with Mono 5.10.0.140, using the included Gtk#

The symptoms are as, as described above, a broken internationalization (the current locale's language mixed with English).

In conclusion: The issue seems to be invalid. Mono 5.10.0.140 does not depend on `libMonoPosixHelper.dll`. So it is not a bug if it is not included in the installer.
However, the fact that internationalization seems to be broken for Gtk# applications should be addressed. I will open a new issue which will include my code sample.

Since I am not sure that I did not miss anything, I leave the issue open.

Thanks.
